sympy/simplify/tests/test_cse.py

```diff
@@ -399,3 +441,151 @@ def test_issue_8891():
         ans = ([(x0, x + y)], [x0, cls([[x0, 0], [0, 0]])])
         assert res == ans
         assert isinstance(res[1][-1], cls)
+
+def test_replace_subsequence():
+    l = [1, 2, 3, 2, 3, 4]
+    replace_subsequence(l, [2, 3], 5)
+    assert l == [1, 5, 5, 4]
+
+    l = [1, 2, 1, 2, 3]
+    replace_subsequence(l, [1, 2, 3], 4)
+    assert l == [1, 2, 4]
+
+    l = [1, 2, 1, 2, 1, 2]
+    replace_subsequence(l, [1, 2], 3)
+    assert l == [3, 3, 3]
+
+
+def test_shortest_repeated_subsequence():
+    a, b, c = 'aabacde', 'aabade', 'abcde'
+    abc = '$'.join([a, b, c])
+    assert shortest_repeated_subsequence(abc, ignore='$') in ['ab', 'de']
+
+    aa = '$'.join([a, a])
+    assert shortest_repeated_subsequence(aa, ignore='$') == a
+
+    ab = '$'.join([a, b])
+    assert shortest_repeated_subsequence(ab, ignore='$') == 'de'
+
+    assert shortest_repeated_subsequence('xxx') == 'xx'
```

Comment:
I'm actually unclear how 'xxx' should be processed. This was the behavior of the algorithm, so I went with it.